modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinClientCodegen.java

```diff
@@ -535,11 +538,16 @@ private void processMultiplatformLibrary(final String infrastructureFolder) {
         importMapping.put("Base64ByteArray", packageName + ".infrastructure.Base64ByteArray");
         importMapping.put("OctetByteArray", packageName + ".infrastructure.OctetByteArray");
 
+        // bundled client file
+        supportingFiles.add(new SupportingFile("ApiClient.kt.mustache", baseFolder, "ApiClient.kt"));
```

Comment:
Note that renaming `ApiClient.kt.mustache` to `ApiClientBase.kt.mustache` is going to cause a lot of problems for anyone who has customized `ApiClient.kt.mustache`.

Can we find a naming strategy which retains `ApiClient.kt.mustache` without breaking existing extensions?

If not, we need to mark this PR as a breaking change and provide some guidelines in the PR for how one would migrate from using multiplatform before 5.0 to use it after this PR is merged.

Reply:
It's breaking pretty much, because it used experimental / alpha state libraries (MPP itself is still experimental) and they had a lot of breaking changes themselves, so that not really something you can avoid. I tried to do my best and migrate it to stable versions so there's no deprecated, opt-in or otherwise experimental stuff anymore but it might be incompatible here and there.

Reply:
I'll see what I can do to add a little migration guide or so